tests/array/tarray.nim

```diff
@@ -1,116 +1,124 @@
 discard """
-output: '''
-[4, 5, 6]
-[16, 25, 36]
-[16, 25, 36]
-apple
-banana
-Fruit
-2
-4
-3
-none
-skin
-paper
-@[2, 3, 4]321
-9.0 4.0
-3
-@[(1, 2), (3, 5)]
-2
-@["a", "new one", "c"]
-@[1, 2, 3]
-3
-dflfdjkl__abcdefgasfsgdfgsgdfggsdfasdfsafewfkljdsfajs
-dflfdjkl__abcdefgasfsgdfgsgdfggsdfasdfsafewfkljdsfajsdf
-kgdchlfniambejop
-fjpmholcibdgeakn
-2.0
-a:1
-a:2
-a:3
-ret:
-ret:1
-ret:12
-123
-'''
 joinable: false
 target: "c cpp js"
+labels: "array"
+description: '''
+  . From https://github.com/nim-lang/Nim/issues/6853
+    Impossible to create an empty const array
+
+  . From https://github.com/nim-lang/Nim/issues/1669
+    Regression: array using high
+    ___
+
+  . From https://github.com/nim-lang/Nim/issues/6852
+    Negative range for array makes no sense but compiles, and last invalid
+    line causes segmentation fault.
+  . Semcheck negative array length:
+    https://github.com/nim-lang/Nim/pull/7518
+    ___
+
+  . From https://github.com/nim-lang/Nim/issues/7153
+    Unsigned integers could not be used as array indexes
+  . Array indexing does not have to be bound by the "ordinal type" constraint.
+    ___
+
+  . From https://github.com/nim-lang/Nim/issues/8049
+    Converter applied when it should not be
+  . Fixed by `introduce precise string '[]', '[]=' accessors`
+    https://github.com/nim-lang/Nim/commit/df4d5b77a14b3b84f3ff37b3741dd039c9e21ce4
+    ___
+
+  . From https://github.com/nim-lang/Nim/issues/6675
+    Wrong indices in arrays not starting with 0
+  . Fixed by
+    https://github.com/nim-lang/Nim/commit/0f5261e9711c3fd57241874963bd5e45b11ed65e
+    ___
+
+  . From https://github.com/nim-lang/Nim/issues/8316
+    BUG: "varargs[string, $]" calls $ n^2 times instead of n times (n=len(varargs))
+'''
 """
 
 block tarray:
+
   type
     TMyArray = array[0..2, int]
     TMyRecord = tuple[x, y: int]
     TObj = object
       arr: TMyarray
 
+  block not_sure_what_this_tests:
 
-  proc sum(a: openarray[int]): int =
-    result = 0
-    var i = 0
-    while i < len(a):
-      inc(result, a[i])
-      inc(i)
-
-  proc getPos(r: TMyRecord): int =
-    result = r.x + r.y
-
-  doAssert sum([1, 2, 3, 4]) == 10
-  doAssert sum([]) == 0
-  doAssert getPos( (x: 5, y: 7) ) == 12
+    proc sum(a: openarray[int]): int =
+      result = 0
+      var i = 0
+      while i < len(a):
+        inc(result, a[i])
+        inc(i)
 
-  # bug #1669
-  let filesToCreate = ["tempdir/fl1.a", "tempdir/fl2.b",
-              "tempdir/tempdir2/fl3.e", "tempdir/tempdir2/tempdir3/fl4.f"]
+    proc getPos(r: TMyRecord): int =
+      result = r.x + r.y
 
-  var found: array[0..filesToCreate.high, bool]
+    doAssert sum([1, 2, 3, 4]) == 10
+    doAssert sum([]) == 0
+    doAssert getPos( (x: 5, y: 7) ) == 12
```

Comment:
I'm moving the openArray to its own block and renaming this block to `leftoverTests`. I'm a fan of letting things that don't really gel together have explicitly bad names, since it eases further discussions about the validity of those.